atomic_reactor/utils/resource_manager.py

```diff
@@ -0,0 +1,132 @@
+"""
+Copyright (c) 2022 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+
+import os
+import subprocess
+import logging
+import json
+from datetime import datetime
+from shlex import quote
+
+ROOT_PATH = 'resource_manager/'
+logger = logging.getLogger(__name__)
+
+
+class ResourceManager:
+    '''
+        Manages build resources
+    '''
+
+    def __init__(self):
+        pass
+
+    def check_avail(self, user, host):
+        try:
+            data = os.popen(f'ssh {user}@{host} -4C "cat {ROOT_PATH}info.json"').read()
+            if data == '':
+                logger.warning("%s hasn't been deployed yet", host)
+                return False
+            else:
+                # parse slot count, arch
+                meta = json.loads(data)
+                logger.debug("%s is up", meta['hostname'])
+
+                # check slots status
+                meta['slots'] = []
+                free_slots = 0
+                i = 0
+                while i < meta.get('max_slot_count', 1):
+                    slot = os.popen(f'ssh {user}@{host} -4C "cat {ROOT_PATH}slot_{i}.json"').read()
+                    if slot == '0\n':
+                        state = 'free'
+                        prid = ''
+                        free_slots += 1
+                    elif slot == "":
+                        logger.warning("host: %s slot: %s has invalid slot state", host, i)
+                        state = 'unknown'
+                        prid = 'unknown'
+                    else:
+                        state = 'running'
+                        logger.debug(slot)
+                        slot_data = json.loads(slot)
+                        prid = slot_data.get("prid", "unknown")
+                    logger.debug("%s slot %s is %s", meta['hostname'], i, state)
+                    meta['slots'].append({'sid': i, 'state': state, 'prid': prid})
+                    i += 1
+                meta['free_slots'] = free_slots
+                return meta
+        except Exception as err:
+            logger.debug(err)
+            logger.error("Availiability check failed fo host: %s", host)
+            return False
+
+    def unlock(self, user, host, slot):
+        logger.debug('Unlocking')
+        retries = 0
+        while retries < 3:
+            cmd = ('ssh %s@%s -C4 "echo \'0\' > %sslot_%s.json"', user, host, ROOT_PATH, slot)
```

Comment:
Is this correct?
```suggestion
            cmd = 'ssh %s@%s -C4 "echo \'0\' > %sslot_%s.json"' % (user, host, ROOT_PATH, slot)
```

Can we use F-strings?